crates/rune/src/ast/path.rs

```diff
@@ -68,3 +72,68 @@ impl<'a> Resolve<'a> for Path {
         Ok(output)
     }
 }
+
+#[derive(Debug, Clone, ToTokens, Spanned)]
+pub enum PathSegment {
+    Ident(ast::Ident),
+    Crate(ast::Crate),
+    Super(ast::Super),
+}
+
+impl PathSegment {
+    /// Borrow as an identifier.
+    ///
+    /// This is only allowed if the PathSegment is `Ident(_)`
+    /// and not `Crate` or `Super`.
+    pub fn try_as_ident(&self) -> Option<&ast::Ident> {
+        if let PathSegment::Ident(ident) = self {
+            Some(ident)
+        } else {
+            None
+        }
+    }
+}
+
+impl Parse for PathSegment {
+    fn parse(parser: &mut Parser<'_>) -> Result<Self, ParseError> {
+        let token = parser.token_peek_eof()?;
+        match token.kind {
+            ast::Kind::Ident(_) => Ok(PathSegment::Ident(parser.parse()?)),
+            ast::Kind::Crate => Ok(PathSegment::Crate(parser.parse()?)),
+            ast::Kind::Super => Ok(PathSegment::Super(parser.parse()?)),
+            _ => {
+                return Err(ParseError::new(
+                    token,
+                    ParseErrorKind::TokenMismatch {
+                        expected: ast::Kind::Ident(ast::StringSource::Text),
+                        actual: token.kind,
+                    },
+                ))
+            }
+        }
+    }
+}
+
+impl Peek for PathSegment {
+    fn peek(t1: Option<ast::Token>, _t2: Option<ast::Token>) -> bool {
+        matches!(peek!(t1).kind, ast::Kind::Ident(_) | ast::Kind::Crate | ast::Kind::Super)
+    }
+}
+
+impl<'a> Resolve<'a> for PathSegment {
```

Comment:
The urge to make this shortcut is strong, but I don't think we should let PathSegment's resolve to strings regardless. I.e. we never want `super` or `crate` to become part of an `Item`. What they do is structural: they change the resolution to the crate root, or move "up" one level when resolving paths.

To see why, consider what would happen here (where `into_components()` is used) if the path contains `super` or `crate`: https://github.com/rune-rs/rune/blob/4d4bb71ba7df2938f60298b743ddd58ccea672a6/crates/rune/src/index.rs#L729